script/tool/lib/src/common/repository_package.dart

```diff
@@ -53,6 +54,44 @@ class RepositoryPackage {
   /// The package's top-level README.
   File get readmeFile => directory.childFile('README.md');
 
+  /// The package's top-level README.
+  File get changelogFile => directory.childFile('CHANGELOG.md');
+
+  /// The package's top-level README.
+  File get authorsFile => directory.childFile('AUTHORS');
+
+  /// The lib directory containing the package's code.
+  Directory get libDirectory => directory.childDirectory('lib');
+
+  /// The test directory containing the package's Dart tests.
+  Directory get testDirectory => directory.childDirectory('test');
+
+  /// Returns the directory containing support for [platform].
+  Directory platformDirectory(FlutterPlatform platform) {
+    late final String directoryName;
+    switch (platform) {
+      case FlutterPlatform.android:
+        directoryName = 'android';
+        break;
+      case FlutterPlatform.ios:
+        directoryName = 'ios';
+        break;
+      case FlutterPlatform.linux:
+        directoryName = 'linux';
+        break;
+      case FlutterPlatform.macos:
+        directoryName = 'macos';
+        break;
+      case FlutterPlatform.web:
+        directoryName = 'web';
+        break;
+      case FlutterPlatform.windows:
+        directoryName = 'windows';
+        break;
+    }
```

Comment:
Please add a default which throws an exception.  I'm pretty sure incomplete switch statements are not an error in Dart, but our linter flags might catch it.

Reply:
We have the lint for that enabled in the repo, so if a new platform is added to the enum CI will fail unless this is updated.